src/plugins/initiatives/createGraph.js

```diff
@@ -89,28 +112,58 @@ export function createWeightedGraph(
       weights.nodeWeights.set(node.address, weight);
     }
 
-    // Generic approach to adding edges when the reference detector has a hit.
+    // Generic approach to adding edges and entries from EdgeSpec or URLs.
     const edgeHandler = (
       edges: $ReadOnlyArray<URL> | EdgeSpec,
-      createEdge: EdgeFactoryT
+      createEdge: EdgeFactoryT,
+      field: ?NodeEntryField
```

Comment:
Why is field optional?

Reply:
Champion URL reference detection uses this same handler, however we don't allow any NodeEntries for it. As NodeEntries describe a _contribution_ while we're expecting _users_.

Looking at this again though. This can be cleaned up. Let me factor it out to a new handler.

Reply:
Separated the handlers, so champions (the previous case where we took null here), just calls a `urlHandler`. And the others call the `edgeSpecHandler`, making this a required argument.